Handle pod deletion event

cqbqdd11519 opened the pull request:
Fix #15
If a pod is terminating (queued to be deleted) until approved/rejected,
Approval CR is marked as Canceled state and the rest logics are skipped

jwkim1993 commented on internal/k8s.go:
```diff
@@ -64,7 +64,7 @@ func UpdateApproval(c client.Client, name types.NamespacedName, result tmaxv1.Re
 	}
 
-	if approval.Status.Result == tmaxv1.ResultWaiting || approval.Status.Result == "" {
+	if approval.Status.Result != tmaxv1.ResultApproved && approval.Status.Result != tmaxv1.ResultRejected {
```
Failed로 끝난 상황에서 pipeline이 지워져도 똑같이 Cancel로 가는건가요?

cqbqdd11519 replied:
이걸 왜 이렇게 바꿨는지 모르겠네요 허허
다시 되돌리겠습니다

cqbqdd11519 committed:
[fix] Handle pod deletion event

Fix #15
If a pod is terminating (queued to be deleted) until approved/rejected,
Approval CR is marked as Canceled state and the rest logics are skipped
